internal/auth/auth.go

```diff
@@ -81,6 +68,29 @@ type State struct {
 	Interval        int    `json:"interval"`
 }
 
+func NewAuthenticaor(audience, id, deviceCodeUrl, tokenUrl string) Authenticator {
```

Comment:
there's no need for a New... method if internally is passing the received arguments directly, without required transformation, validation or initialization with default values. I'd suggest export (uppercase) those 4 properties (if needed) and use &Authenticator{...} where needed.